src/Dropdown/Dropdown.js

```diff
@@ -97,36 +156,66 @@ export default class Dropdown extends Util.mixin(BindMixin) {
 
   handleMenuToggle(e) {
     e.stopPropagation();
+    let menuDirection = this.state.menuDirection;
+
+    // If the menu isn't open, then we're about to open it and we need to
+    // calculate the direction every time.
+    if (!this.state.isOpen) {
+      menuDirection = this.getMenuDirection(this.state.menuHeight);
+    }
+
     this.setState({
+      menuDirection,
       isOpen: !this.state.isOpen
     });
   }
 
   render() {
+    // Set a key based on the menu height so that React knows to keep the
+    // the DOM element around while we are measuring it.
+    let dropdownKey = this.state.knowMenuHeight || 'initial-render';
+    let dropdownMenu = null;
+    let dropdownMenuClassSet = classNames(
+      this.state.menuDirection,
+      this.props.dropdownMenuClassName
+    );
     let dropdownStateClassSet = {
       'open': this.state.isOpen
     };
-    let dropdownMenu = null;
     let items = this.props.items;
+    let transitionName =
+      `${this.props.transitionName}-${this.state.menuDirection}`;
     let wrapperClassSet = classNames(
+      this.state.menuDirection,
```

Comment:
#nit can we get `state` and `props` as variables, so we don't access them multiple times?